### ISSUE TITLE
2D ugrid reader request

### ISSUE THREAD

**Issue by tuckerbabcock** [SCOREC/core#281]
I'd like to request support for reading a 2D ugrid mesh. The documentation of the 2D ugrid format is here: http://www.simcenter.msstate.edu/software/downloads/doc/ug_io/2d_grid_file_type_ugrid.html
 
I'll be happy to work on this.

**Comment by tuckerbabcock** [comment]
@cwsmith  I think I’ve finished modifying the ugrid reader to read the 2D mesh format, but after reading, I try to verify the mesh with: `ugrid_mesh->verify();`, which fails because I haven’t classified mesh vertices onto any model entities. The 2D ugrid format classifies both triangles/quads onto model faces, and the boundary edges onto model edges, yet doesn’t provide classification information for the mesh vertices. What am I to do? I’ve tried to follow the existing approach for reading a 3D ugrid mesh, but it doesn’t seem like that explicitly classifies mesh vertices. I suppose the classification of every vertex could be derived from the classification of adjacent edges, though this seems tedious.
Will I still be able to use MeshAdapt without this vertex classification?

Thanks

**Comment by cwsmith** [comment]
The 3D ugrid reader creates a very simple model using the `deriveMdsModel(...)` API
https://github.com/SCOREC/core/blob/79032805c8c1ce90d661130df11b811387c24b4d/test/ugrid.cc#L60
described here:
https://github.com/SCOREC/core/wiki/Geometric-Models#gmi-topological-model

I don't see any mesh classification data being set in `mdsUgrid.cc`:
https://github.com/SCOREC/core/blob/59d03165714ac2c2069f4de6a3d08bd3ba041d20/mds/mdsUgrid.cc

Would you please share your code? A link to your fork/branch would be ideal.

**Comment by tuckerbabcock** [comment]
Hi Cameron,

I've been working here: https://github.com/tuckerbabcock/core/blob/ugrid2D-dev/mds/mdsUgrid.cc

It's fairly rough right now, but I've added a load ugrid function with the signature
`Mesh2* loadMdsFromUgrid(gmi_model* g, const char* filename, const int mesh_dim)`,
which calls the existing loadMdsFromUgrid function if 3D, and if 2D calls a new function I wrote `readUgrid2D`. 

In functions called by `readUgrid2D` I iterate over and classify every mesh entity. To classify the vertices I've used the classification of the adjacent edges, done in `classifyVtx`. After I manually classified every entity the mesh passes verification.

I'll clean up the interface, and remove the additional function I introduced to load the mesh and instead will deduce the mesh dimension from the information contained in the mesh header.
